File: app/src/main/java/com/hazizz/droid/Communication/responsePojos/commentSectionPojos/PojoComment.java
package com.hazizz.droid.Communication.responsePojos.commentSectionPojos;

import com.hazizz.droid.Communication.responsePojos.Pojo;
import com.hazizz.droid.Communication.responsePojos.taskPojos.PojoCreator;

import lombok.Data;

@Data
public class PojoComment implements Pojo {
    private int id;

    private String content;

    private PojoCreator creator;

    private PojoComment[] children;

    private String creationDate;
    private boolean moreChildren;

    public PojoComment(int id, String content, PojoCreator creator, PojoComment[] children, String creationDate, boolean moreChildren) {
        this.id = id;
        this.content =  content;
        this.creator = creator;
        this.children = children;
        this.creationDate = creationDate;
        this.moreChildren = moreChildren;
    }
}

File: lib/communication/pojos/pojo_comment.dart
import 'package:json_annotation/json_annotation.dart';

import 'Pojo.dart';
import 'PojoCreator.dart';

part 'pojo_comment.g.dart';

@JsonSerializable()
class PojoComment  implements Pojo {
  String content;
  DateTime creationDate;
  PojoCreator creator;
  bool hiddenChildren;
  int id;
  List<PojoComment> children;

  PojoComment({this.content, this.creationDate, this.creator, this.hiddenChildren, this.id, this.children});

  @override
  factory PojoComment.fromJson(Map<String, dynamic> json) =>
      _$PojoCommentFromJson(json);

  @override
  Map<String, dynamic> toJson() => _$PojoCommentToJson(this);
}